File: B트랙/8주차/참고 코드/floyd-warshall.java
import java.io.*;
import java.util.*;

public class Main
{
    public static void main(String[] args) throws IOException // https://www.acmicpc.net/problem/11404
    {
        BufferedReader scan = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(scan.readLine());
        int n = Integer.parseInt(st.nextToken());
        st = new StringTokenizer(scan.readLine());
        int m = Integer.parseInt(st.nextToken());
        long[][] graph = new long[n+1][n+1];
        for (int i = 1; i <= n; i++)
        {
            Arrays.fill(graph[i],Long.MAX_VALUE);
            graph[i][i] = 0;
        }
        for (int i = 0; i < m; i++)
        {
            st = new StringTokenizer(scan.readLine());
            int v1 = Integer.parseInt(st.nextToken());
            int v2 = Integer.parseInt(st.nextToken());
            long value = Long.parseLong(st.nextToken());
            graph[v1][v2] = Math.min(graph[v1][v2], value);
        }
        for (int k = 1; k <= n; k++)
        {
            for (int i = 1; i <= n; i++)
            {
                for (int j = 1; j <= n; j++)
                {
                    if (graph[i][k] != Long.MAX_VALUE && graph[k][j] != Long.MAX_VALUE)
                    {
                        graph[i][j] = Math.min(graph[i][j], graph[i][k] + graph[k][j]);
                    }
                }
            }
        }
        StringBuilder ans = new StringBuilder();
        for (int i = 1; i <= n; i++)
        {
            for (int j = 1; j <= n; j++)
            {
                ans.append((graph[i][j] == Long.MAX_VALUE) ? 0 : graph[i][j]).append(' ');
            }
            ans.append('\n');
        }
        System.out.println(ans);
        scan.close();
    }
}

File: B트랙/8주차/참고 코드/floyd-warshall.py
# https://www.acmicpc.net/problem/11404
import sys
input = sys.stdin.readline
n = int(input().strip())
m = int(input().strip())
graph = [[float('inf') for __ in range(n+1)] for _ in range(n+1)]
for i in range(1,n+1):
    graph[i][i] = 0
for i in range(m):
    v1, v2, value = map(int,input().strip().split())
    graph[v1][v2] = min(graph[v1][v2], value)
for k in range(1, n + 1):
    for i in range(1, n + 1):
        for j in range(1, n + 1):
            if graph[k][j] != float('inf') and graph[i][k] != float('inf'):
                graph[i][j] = min(graph[i][j], graph[i][k] + graph[k][j])
for i in range(1, n+1):
    for j in range(1, n+1):
        if graph[i][j] == float('inf'):
            print(0, end = ' ')
        else:
            print(graph[i][j], end = ' ')
    print()
